pkg/pgmodel/ingestor/trace/writer.go

```diff
@@ -313,29 +323,43 @@ func (t *traceWriterImpl) InsertTraces(ctx context.Context, traces pdata.Traces)
 		}
 	}
 
+	metrics.InsertBatchSize.With(traceSpanLabel).Observe(float64(spanBatch.Len()))
+	metrics.InsertBatchSize.With(traceEventLabel).Observe(float64(eventBatch.Len()))
+	metrics.InsertBatchSize.With(traceLinkLabel).Observe(float64(linkBatch.Len()))
+
 	start := time.Now()
-	if err := t.sendBatches(ctx, eventBatch, linkBatch, spanBatch); err != nil {
-		return fmt.Errorf("error sending trace batches: %w", err)
+	if err := t.sendBatch(ctx, eventBatch); err != nil {
+		return fmt.Errorf("error sending trace event batch: %w", err)
+	}
+	metrics.InsertDuration.With(traceEventLabel).Observe(time.Since(start).Seconds())
```

Comment:
when this PR is rebased, there is a label `code`, that denotes success or failure. success has `2xx` and failure has `500`. We should make this and the below `.Observe()` as deferable and based on the condition, duration for a success series of a failure series should be set.